tool/tsh/db.go

```diff
@@ -109,6 +113,95 @@ func onListDatabases(cf *CLIConf) error {
 	return trace.Wrap(showDatabases(cf.SiteName, databases, activeDatabases, roleSet, cf.Format, cf.Verbose))
 }
 
+type databaseListing struct {
+	Proxy    string           `json:"proxy"`
+	Cluster  string           `json:"cluster"`
+	roleSet  services.RoleSet `json:"-"`
+	Database types.Database   `json:"database"`
+}
+
+func listDatabasesAllClusters(cf *CLIConf) error {
+	profile, profiles, err := client.Status(cf.HomePath, "")
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	if profile != nil {
+		profiles = append(profiles, profile)
+	}
+
+	dbListings := make([]databaseListing, 0)
+	for _, p := range profiles {
+		cf.Proxy = p.ProxyURL.Host
+		tc, err := makeClient(cf, true)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+
+		result, err := tc.ListDatabasesAllClusters(cf.Context, nil /* custom filter */)
+		// Don't make the user re-login to every proxy.
+		if client.IsExpiredCredentialError(err) {
```

Comment:
ProfileStatus has `IsExpired` method, can it be used to gracefully skip over expired profiles?